content/en/blog/_posts/2023-09-20-kubeadm-use-etcd-learner-mode.md

```diff
@@ -0,0 +1,81 @@
+---
+layout: blog
+title: 'Kubeadm: Use ETCD Learner to join a control plane node safely'
+date: 2023-09-20
+slug: kubeadm-use-etcd-learner-mode
+---
+
+**Author:** Paco Xu(DaoCloud)
+
+Kubeadm now supports ETCD learner mode, which allows you to enhance the resilience and stability
+of your Kubernetes clusters by leveraging the [learner mode](https://etcd.io/docs/v3.4/learning/design-learner/#appendix-learner-implementation-in-v34) feature introduced in ETCD version 3.4.
+This guide will walk you through using ETCD learner mode with kubeadm.
+
+In v1.27, kubeadm introduced a new feature gate `EtcdLearnerMode`. With this feature gate enabled,
+when joining a new control plane node, a new etcd member will be created as a learner and 
+promoted to a voting member only after the etcd data are fully aligned.
+
+## What are the advantages of using ETCD learner mode?
+
+Etcd learner mode offers several compelling reasons to consider its adoption
+in Kubernetes clusters:
+
+1. **Enhanced Resilience**: Etcd learner nodes are non-voting members that catch up with
+the leader's logs before becoming fully operational. This prevents new cluster members
+from disrupting the quorum or causing leader elections, making the cluster more resilient
+during membership changes.
+2. **Reduced Cluster Unavailability**: Traditional approaches to adding new members often
+result in cluster unavailability periods, especially in slow infrastructure or misconfigurations.
+Etcd learner mode minimizes such disruptions.
+3. **Simplified Maintenance**: Learner nodes provide a safer and reversible way to add or replace
+cluster members. This reduces the risk of accidental cluster outages due to misconfigurations or
+missteps during member additions.
+4. **Improved Network Tolerance**: In scenarios involving network partitions, learner mode allows
+for more graceful handling. Depending on the partition a new member lands, it can seamlessly
+integrate with the existing cluster without causing disruptions.
```

Comment:
```suggestion
   for more graceful handling. Depending on the partition a new member lands, it can seamlessly
   integrate with the existing cluster without causing disruptions.
```